perimeter/shared_access.sp

```diff
@@ -895,3 +911,761 @@ control "rds_db_snapshot_shared_with_trusted_accounts" {
     service = "AWS/RDS"
   })
 }
+
+locals {
+  resource_policy_shared_account_sql = <<-EOT
+    select
+      r.__ARN_COLUMN__ as resource,
+      case
+        when jsonb_array_length(pa.allowed_principal_account_ids) = 0 then 'ok'
+        when pa.access_level = 'private' then 'ok'
+        when pa.access_level = 'public' then 'alarm'
+        when pa.allowed_principal_account_ids <@ to_jsonb(($1)::text[]) then 'ok'
+        else 'alarm'
+      end as status,
+      case
+        when jsonb_array_length(pa.allowed_principal_account_ids) = 0 then title || ' trust policy does not reference any accounts.'
+        when pa.access_level = 'private' then title || ' trust policy does not reference any cross accounts.'
+        when pa.access_level = 'public' then title || ' trust policy allows access to all AWS accounts.'
+        when pa.allowed_principal_account_ids <@ to_jsonb(($1)::text[]) then title || ' trust policy allows cross account access to trusted accounts.'
+        else title || ' trust policy contains ' || count(pa.shared_statement_ids) || ' statement(s) that allows cross account access to untrusted accounts.'
```

Comment:
Can you please try using a reason like the one in `control.ec2_ami_shared_with_trusted_accounts`? I think it'd be helpful to show some account IDs if we can, so instead of a user running this control, seeing untrusted accounts have access (but not knowing which ones), they would know a 1-2 untrusted account IDs from the reason, and this may be enough if there are only 1-2 untrusted accounts overall.

Reply:
If this updated reason format provides more info and fits on screen still, we should adapt the other queries to use it as well.